### PR TITLE
Addition of a new display to see the server name easily

### PR DESCRIPTION
**Pull request by Jilijou**
Adding a new object displaying the current profile name.
NEW :
- getHx_name() > new function + reference
- hx_name-placeholder & fe-top-nav__bar-right-item > new list object + css style

MODIFIED :
- .fe-top-nav__topnav .fe-top-nav__bar > css : Added padding right
- profile() > Added a different return when logged in
- .hxtool_topnav_mainbutton:hover > Modified so it only affects the buttons

image

### CODE REVIEW

**Comment by Jilijou** [comment]
Note :
This pull request is here to answer to this enhancement request
https://github.com/fireeye/HXTool/issues/103

**Comment by B0fH** [comment]
Hi @JilianJouault -
 Thank you for posting this, much appreciated! May I suggest that it might make more sense to store the profile name in session[] as opposed to querying the /profile API on every page load. You could simply add (to this block here https://github.com/fireeye/HXTool/blob/master/hxtool.py#L505):
`
session["profile_name"] = ht_profile["hx_name"]
`
and in your Javascript
`
$('#hx_name-placeholder').text({{session['profile_name']}});
`

Thanks,
 Elazar

File: hxtool_api.py [hunk 1]
```diff
@@ -3367,7 +3367,10 @@ def chartjs_inactive_hosts_per_hostset(hx_api_object):
 def profile():
 	if request.method == 'GET':
 		profiles = hxtool_global.hxtool_db.profileList()
-		return json.dumps({'data_count' :  len(profiles), 'data' : profiles})
+		if 'ht_hx_name' in session:
+			return json.dumps({'data' : profiles, 'cur_hx_name' : session['ht_hx_name']})
+		else:
+			return json.dumps({'data_count' :  len(profiles), 'data' : profiles})
```
Comment (B0fH) [inline hxtool_api.py:3373]: Remove this change altogether as we're now storing the profile name in session

File: templates/layout.html [hunk 2]
```diff
@@ -41,6 +41,15 @@
...
 	<script>
+		
+		//Function to get and show the name of the current server
+		function getHx_name() {
+			hxtool_ajax_get_request("/api/v1/profile", "", function(myprofile) {
+				var n = 0;
+				var hx_name = myprofile.cur_hx_name;
+				$('#hx_name-placeholder').text(hx_name);
+			});
+		}
```
Comment (B0fH) [inline templates/layout.html:52]: You can remove this as well.

File: templates/layout.html [hunk 3]
```diff
@@ -94,7 +103,7 @@
 		}
 		
 		$(document).ready(function() {
-
+			getHx_name();
```
Comment (B0fH) [inline templates/layout.html:106]: Change this to:
`
$('#hx_name-placeholder').text({{session['ht_hx_name']}});
`

**Review by B0fH — CHANGES_REQUESTED** [review]
Hi @Jilijou -
 Kindly see my comments below. Once you've added the profile name to session, you can do away with the API call code, as I've noted below.

Thanks,
 Elazar

**Comment by Jilijou** [comment]
Hello @B0fH,
I just made the modification and it still works 🎉 Proof that I did not fully understand how requests work... I'll need to make a bit more research on that.

Hope it's all good now,
Jilian

**Review by B0fH — APPROVED** [review]
Looks good, thank you!

**Pull request merged**